File: MPP/Curs2/Curs2/Java_tasks/tasks/model/src/main/java/tasks/repository/IRepository.java
package tasks.repository;

/**
 * Created by grigo on 11/14/16.
 */
public interface IRepository<ID, T> {
    int size();
    void save(T entity);
    void delete(ID id);
    void update(ID id, T entity);
    T findOne(ID id);
    Iterable<T> findAll();
}

File: MPP/Client_Java_Server_C#/ServerC#/ConcursServer/Persistence/IRepository.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Concurs.repository
{
    public interface IRepository<ID,E>
    {
        E FindOne(ID id);

        IEnumerable<E> FindAll();

        ID Save(E entity);

        void Delete(ID id);

        void Update(ID id, E entity);

        int Size();
    }
}
